core/src/commonMain/kotlin/com/ustadmobile/core/viewmodel/login/LoginViewModel.kt

```diff
@@ -110,12 +110,18 @@ class LoginViewModel(
     private val parseCredentialUsernameUseCase: ParseCredentialUsernameUseCase by instance()
 
     init {
-        nextDestination = savedStateHandle[UstadView.ARG_NEXT] ?: ClazzListViewModel.DEST_NAME_HOME
 
         serverUrl = savedStateHandle[UstadView.ARG_LEARNINGSPACE_URL]
             ?: apiUrlConfig.presetLearningSpaceUrl ?: "http://localhost"
         savedStateHandle[UstadView.ARG_LEARNINGSPACE_URL] = serverUrl
 
+        val getDefaultDestinationUseCase: GetDefaultDestinationUseCase =
+            di.on(LearningSpace(serverUrl)).direct.instance()
+        viewModelScope.launch {
+            nextDestination = savedStateHandle[UstadView.ARG_NEXT] ?:
```

Comment:
This is a race condition - no gaurantee that it will be ready first. You could use CompleteableDeferred.